src/lib/homestore/heap_chunk_selector.cpp

```diff
@@ -0,0 +1,38 @@
+#include "heap_chunk_selector.h"
+
+namespace homeobject {
+
+void HeapChunkSelector::add_chunk(csharedChunk& chunk) {
+    uint32_t pdevId = VChunk(chunk).get_pdev_id();
+    auto it = _pdev_heap_map.find(pdevId);
+    if (it == _pdev_heap_map.end()) it = _pdev_heap_map.try_emplace(pdevId, std::make_shared<std::pair<std::mutex, VChunkHeap>>()).first;
```

Comment:
Can we use the pattern:
```
if (auto [it, _] = _pdev_heap_map.try_emplace(...); _pdev_heap_map.end() != it) {
    auto l = std::scoped_lock(it->second->first);
    it->second->second.emplace(chunk);
} else RELEASE_ASSERT(false, "Something Bad!");
```
then you don't have to call `find` as well.

Reply:
good, this pattern seems better, thanks for the suggestion